File: src/app/Class/ArtistInBand.ts
import { Artist } from './Artist';
import { Band } from './Band';

export class ArtistInBand {
  id: number;
  artist: Artist;
  band: Band;
  activityStart: Date;
  activityEnd: Date;
  isActive: boolean;
  roles: string[];

  constructor(obj?: any) {
    this.id = (obj && obj.id) || 0;
    this.artist = (obj && obj.artist) || null;
    this.band = (obj && obj.band) || null;
    this.activityStart = (obj && obj.activityStart) || null;
    this.activityEnd = (obj && obj.activityEnd) || null;
    this.isActive = (obj && obj.active) || null;
    this.roles = (obj && obj.roles) || [];
  }

  // Getters

  getId(): number {
    return this.id;
  }

  getArtist(): Artist {
    return this.artist;
  }

  getBand(): Band {
    return this.band;
  }

  getActivityStart(): Date {
    return this.activityStart;
  }
  getActivityEnd(): Date {
    return this.activityEnd;
  }

  getIsActive(): boolean {
    return this.isActive;
  }

  getRoles(): string[] {
    return this.roles;
  }
}

File: src/main/java/musiccube/entities/ArtistInBand.java
package musiccube.entities;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

import javax.persistence.*;
import java.util.Date;

@JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
@Entity
public class ArtistInBand {

    @Id
    @GeneratedValue
    @Column
    private int id;

    @ManyToOne
    private Artist artist;

    @ManyToOne
    private Band band;

    @Column
    private Date activityStart;

    @Column
    private Date activityEnd;

    @Column
    private boolean isActive;

    @Column(length = 1024)
    private String[] roles;

    public ArtistInBand() {}

    public ArtistInBand(Artist artist, Band band, Date activityStart, Date activityEnd, boolean isActive, String[] roles) {
        this.artist = artist;
        this.band = band;
        this.activityStart = activityStart;
        this.activityEnd = activityEnd;
        this.isActive = isActive;
        this.roles = roles;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public Artist getArtist() {
        return artist;
    }

    public void setArtist(Artist artist) {
        this.artist = artist;
    }

    public boolean isActive() {
        return isActive;
    }

    public void setActive(boolean active) {
        isActive = active;
    }

    public Band getBand() {
        return band;
    }

    public void setBand(Band band) {
        this.band = band;
    }

    public Date getActivityStart() {
        return activityStart;
    }

    public void setActivityStart(Date activityStart) {
        this.activityStart = activityStart;
    }

    public Date getActivityEnd() {
        return activityEnd;
    }

    public void setActivityEnd(Date activityEnd) {
        this.activityEnd = activityEnd;
    }

    public String[] getRoles() {
        return roles;
    }

    public void setRoles(String[] roles) {
        this.roles = roles;
    }
}
